app/models/user.rb

```diff
@@ -0,0 +1,24 @@
+require 'bcrypt'
+
+class User
+  include DataMapper::Resource
+
+  has n, :peeps
+
+  property :id, Serial
+  property :email, String
+  property :password_digest, Text
+
+  def password=(password)
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  def self.authenticate(email, password)
```

Comment:
Would it be possible to rewrite this method to use a guard clause that raises an error if the password is wrong? It might make the control flow even clearer.